ledger/applications.go

```diff
@@ -119,6 +119,10 @@ func newAppLedger(balances transactions.Balances, acctWhitelist []basics.Address
 	}
 
 	for _, aidx := range appGlobalWhitelist {
+		if aidx == 0 {
```

Comment:
Hm, do we actually need 0 to be invalid here? I can't think of a reason why -- it seems like an arbitrary constraint. If we to do want this, there should probably also be a `WellFormed` check on `ForeignApps` so that we can give a more useful error message like `cannot have 0 value in txn.ForeignApps`.

Reply:
we have `appIdx=0` special meaning. So need to be consistent.

Reply:
But the special meaning of 0 is to be `== appIdx`. And since passing appIdx in the whitelist isn't invalid, it seems like this shouldn't be either

Reply:
Not sure I follow. `newAppLedger` does not allow `appIdx == 0`. I think zero appIdx and TEAL abstraction, and should be handled in APIs accepting appIdx that are `AppGlobalState` and `AppLocalState`, and they both have appIdx == 0 check.
Having that `appGlobalWhitelist` with zero is not invalid but it is a value that would be never accessed.

Reply:
> Having that appGlobalWhitelist with zero is not invalid but it is a value that would be never accessed.

It's true that it would never be accessed or used, but it means that anyone creating a transaction now has to validate that all `ForeignApps` values are all nonzero or their transaction will fail. And if I ever want a placeholder value in `ForeignApps` now, I have to use a nonzero value.

I just don't see a reason to put a constraint on it since it doesn't affect the correctness of the code and seems like it could cause confusing failures/error messages later.

Reply:
removed that check